test/testCleanText.js

```diff
@@ -0,0 +1,181 @@
+    });
+
+    context('stripping periods from email addresses', function() {
+      it('should replace a single period', function() {
+        var ts = TextStatistics('textstatistics@example.com');
+        assert.equal(ts.text, 'textstatistics@examplecom.');
+      });
+
+      it('should replace a single period in the first part', function() {
+        var ts = TextStatistics('text.statistics@example.com');
+        assert.equal(ts.text, 'textstatistics@examplecom.');
+      });
+
+      it('should replace two periods in the first part', function() {
+        var ts = TextStatistics('text.stat.istics@example.com');
+        assert.equal(ts.text, 'textstatistics@examplecom.');
+      });
+
+      it('should replace periods with a subdomain', function() {
+        var ts = TextStatistics('textstatistics@test.example.com');
+        assert.equal(ts.text, 'textstatistics@testexamplecom.');
+      });
+
+      it('should replace periods with a subdomain and before the @', function() {
+        var ts = TextStatistics('text.stat.istics@test.example.com');
+        assert.equal(ts.text, 'textstatistics@testexamplecom.');
+      });
+    });
+
+    context('replacing non-terminator punctuation', function() {
+      it('should replace commas with spaces', function() {
+        var ts = TextStatistics('Hello, hi, friend.');
+        assert.equal(ts.text, 'Hello hi friend.');
+      });
+      
+      it('should replace colons with spaces', function() {
+        var ts = TextStatistics('Hello: hi: friend.');
+        assert.equal(ts.text, 'Hello hi friend.');
+      });
+      
+      it('should replace semicolons with spaces', function() {
+        var ts = TextStatistics('Hello; hi; friend.');
+        assert.equal(ts.text, 'Hello hi friend.');
+      });
+      
+      it('should replace parentheses with spaces', function() {
+        var ts = TextStatistics('(Hello (hi) friend).');
+        assert.equal(ts.text, 'Hello hi friend.');
+      });
+      
+      it('should replace slashes with spaces', function() {
+        var ts = TextStatistics('Hello/hi/friend.');
+        assert.equal(ts.text, 'Hello hi friend.');
+      });
+      
+      it('should replace double hyphens with spaces', function() {
+        var ts = TextStatistics('Hello--hi--friend.');
+        assert.equal(ts.text, 'Hello hi friend.');
+      });
+
+      it('should not replace a single dash with spaces', function() {
+        var ts = TextStatistics('Hi-di-ho friend-person!');
+        assert.equal(ts.text, 'Hi-di-ho friend-person.');
+      });
+
+      it('should replace pluses with spaces', function() {
+        var ts = TextStatistics('Hello + hi+friend.');
+        assert.equal(ts.text, 'Hello hi friend.');
+      });
+
+      it('should replace ampersands with spaces', function() {
+        var ts = TextStatistics('Hello&hi & friend.');
+        assert.equal(ts.text, 'Hello hi friend.');
+      });
+
+      it('should replace em-dash with spaces'); // can I do that?
```

Comment:
Em dash is usually used as a word terminator in my experience, rather than to create hyphenated pairs, so I don't see why not!